### PR TITLE
feature/15/create and integrate doc panel

### PR DESCRIPTION
**Pull request by r-sukanya**

### CODE REVIEW

File: client/js/app.js [hunk 1]
```diff
@@ -13,6 +13,36 @@ function loadJSON(endpoint, type, body, callback, error) {
     }
 }
 
+function loadDatasets(endpoint, type, callback, error) {
```
Comment (sebmarste) [inline client/js/app.js]: please merge this function with the generic loadJSON function (maybe the you should always call `callback(response)` )

File: client/js/app.js [hunk 2]
```diff
@@ -38,8 +68,114 @@ function setLoading(loading) {
     }
 }
 
-window.onload = function () {
+function getColumnDataSuccess(response) {
```
Comment (sebmarste) [inline client/js/app.js]: i would rename this to `createColumnList(columns)`

File: client/js/app.js [hunk 3]
```diff
@@ -38,8 +68,114 @@ function setLoading(loading) {
...
+        let tr = document.createElement("tr");
+        let td1 = document.createElement("td");
+        td1.setAttribute("class", "mdl-data-table__cell--non-numeric");
+        td1.append(key.replace(/_/g, " "));
+        tr.appendChild(td1);
+        let td2 = document.createElement("td");
+        td2.setAttribute("class", "mdl-data-table__cell--non-numeric");
+        td2.append(data[key].type);
+        tr.appendChild(td2);
+        table.appendChild(tr);
```
Comment (sebmarste) [inline client/js/app.js]: i would extract all of this to a function `createColumnEntry(column)`

File: client/js/app.js [hunk 4]
```diff
@@ -38,8 +68,114 @@ function setLoading(loading) {
...
+    for (var i = 0; i < ul.children.length; i++) {
+        if (ul.children[i].dataset.selected === 'true') {
+            var dataset = ul.children[i].dataset.val;
+            break;
+        }
+    }
+    generateGraph(dataset, example.textContent)
+}
+
+function getExampleDataSuccess(response) {
```
Comment (sebmarste) [inline client/js/app.js]: maybe rename to `createExampleList(examples)`

File: client/js/app.js [hunk 5]
```diff
@@ -38,8 +68,114 @@ function setLoading(loading) {
...
+    data.forEach(d => {
+        let tr = document.createElement("tr");
+        let td = document.createElement("td");
+        td.setAttribute("class", "mdl-data-table__cell--non-numeric");
+        td.append(d.input);
+        tr.onclick = function () {
+            onExampleSelected(this);
+        };
+        tr.appendChild(td);
+        table.appendChild(tr);
```
Comment (sebmarste) [inline client/js/app.js]: extract all to a function `createExampleEntry(example)`

File: client/js/app.js [hunk 6]
```diff
@@ -38,8 +68,114 @@ function setLoading(loading) {
...
+
+function onExampleSelected(example) {
+    console.log(example.textContent);
+    let ul = document.getElementById("dataset-list");
+    for (var i = 0; i < ul.children.length; i++) {
+        if (ul.children[i].dataset.selected === 'true') {
+            var dataset = ul.children[i].dataset.val;
+            break;
+        }
+    }
```
Comment (sebmarste) [inline client/js/app.js]: could you extract this to a function `getCurrentDataset()` and use this function in all places where a `dataset` is needed?

File: client/js/app.js [hunk 7]
```diff
@@ -38,8 +68,114 @@ function setLoading(loading) {
...
+        ul.appendChild(li);
+    });
+    ul.firstElementChild.dataset.selected = true;
+    getAdditionalData(ul.firstElementChild.dataset.val);
+}
+
+function loadDockData() {
+    loadDatasets("datasets", "GET", loadDatasetSuccess, () => {
+        // TODO handle error
+        setLoading(false)
```
Comment (sebmarste) [inline client/js/app.js]: this should be always in the same function where `setLoading(true)` is called

File: client/public/index.html [hunk 8]
```diff
@@ -27,21 +27,36 @@
             <input type="hidden" value="" name="dataset">
             <i class="mdl-icon-toggle__label material-icons">keyboard_arrow_down</i>
             <label for="dataset" class="mdl-textfield__label">Dataset</label>
-            <ul for="dataset" class="mdl-menu mdl-menu--bottom-left mdl-js-menu" id="dataset-list">
```
Comment (sebmarste) [inline client/public/index.html]: the `for="dataset"` indicates, that the dropdown will always put the current value into the hidden input field (three lines above) 
try to set this up correctly (see [getmdl-select doku](https://github.com/CreativeIT/getmdl-select#data-val))

File: client/js/app.js (hunk 7 above)
Comment (r-sukanya) [reply client/js/app.js]: I dont understand your point. loadDockData is being called on window.onload

File: client/public/index.html (hunk 8 above)
Comment (r-sukanya) [reply client/public/index.html]: intelliJ suggested that for is an invalid attribute.

File: client/js/app.js [hunk 9]
```diff
@@ -38,8 +68,114 @@ function setLoading(loading) {
...
+}
+
+function getExampleDataSuccess(response) {
+    let data = response.examples;
+    let keys = Object.keys(data);
+    let table = document.getElementById('example_body');
+    while (table.firstChild) {
+        table.removeChild(table.firstChild);
+    }
+    data.forEach(d => {
```
Comment (sebmarste) [inline client/js/app.js]: rename `data` to `examples` and `d` to `example`

**Pull request merged**